File: 프로그래머스/3/152995. 인사고과/인사고과.js
function solution(scores) {
    const [wanhoA, wanhoB] = scores[0];
    const wanhoSum = wanhoA + wanhoB;

    // 근무태도 내림차순, 동료평가 오름차순
    scores.sort((a, b) => {
        if (a[0] !== b[0]) return b[0] - a[0];
        return a[1] - b[1];
    });

    let maxB = 0;
    let rank = 1;

    for (const [a, b] of scores) {
        // 지배당한 경우
        if (b < maxB) {
            if (a === wanhoA && b === wanhoB) {
                return -1;
            }
            continue;
        }

        maxB = b;

        if (a + b > wanhoSum) {
            rank++;
        }
    }

    return rank;
}

File: 프로그래머스/3/152995. 인사고과/인사고과.py
def solution(scores):
    wanho_a, wanho_b = scores[0]
    wanho_sum = wanho_a + wanho_b

    # 근무태도 내림차순, 동료평가 오름차순
    scores.sort(key=lambda x: (-x[0], x[1]))

    max_b = 0
    valid_sums = []

    for a, b in scores:
        if b < max_b:
            # 지배당함
            if a == wanho_a and b == wanho_b:
                return -1
            continue
        max_b = b
        valid_sums.append(a + b)

    # 석차 계산
    rank = 1
    for s in sorted(valid_sums, reverse=True):
        if s > wanho_sum:
            rank += 1

    return rank
